### PR TITLE
Everywhere: Add RISC-V 64 target to the build system

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
This is a minimal set of changes to allow `serenity.sh build riscv64` (GCC and Clang) to successfully generate the build environment and start compiling and building. This includes some, but not all, assembly stubs that will be needed later on; they are currently empty. The intent behind submitting this as-is is so that others can easily look at the compile failures and start stubbing out code and functionality more properly.

### CODE REVIEW

File: Userland/Libraries/LibC/arch/riscv64/crti.S
Comment (BertalanD) [inline Userland/Libraries/LibC/arch/riscv64/crti.S:1]: Per the discussion in https://discord.com/channels/830522505605283862/1139481927594803260/1139524004852539523, could you mention that this file is intentionally empty?

File: Userland/Libraries/LibC/arch/riscv64/crtn.S
Comment (BertalanD) [inline Userland/Libraries/LibC/arch/riscv64/crtn.S:1]: Same here.

File: CMakeLists.txt [hunk 1]
```diff
@@ -179,6 +179,10 @@ if("${SERENITY_ARCH}" STREQUAL "aarch64")
     add_compile_options(-Wno-type-limits)
 endif()
 
+if("${SERENITY_ARCH}" STREQUAL "riscv64")
+    add_compile_options(-march=rv64gc)
```
Comment (ADKaster) [inline CMakeLists.txt:183]: I haven't been paying quite enough attention in the #riscv channel as I probably should, but why can't we set this in the toolchain file?

File: Userland/Libraries/LibC/arch/riscv64/setjmp.S
Comment (ADKaster) [inline Userland/Libraries/LibC/arch/riscv64/setjmp.S:1]: These other files are intentionally empty, should they have the same comment?

**Comment by ADKaster** [comment]
Nothing particularly crazy here, just a couple comments

File: CMakeLists.txt (hunk 1 above)
Comment (kleinesfilmroellchen) [reply CMakeLists.txt:183]: If you mean the [...]Toolchain.txt.in files, we *could*, but we want to enforce using the same ISA (and even ISA spec version, if it becomes relevant) on both toolchains.

Comment (ADKaster) [reply CMakeLists.txt:183]: Sure, and we can do that by having the same contents in both files and a comment saying "oy m8, update the other one". Putting it in the toolchain file also makes sure that any CMake-using ports use the same arch flags as the OS itself.

Comment (kleinesfilmroellchen) [reply CMakeLists.txt:183]: Ah, didn't realize that ports also pick this up.

File: Toolchain/CMake/ClangToolchain.txt.in [hunk 2]
```diff
@@ -40,3 +40,8 @@ set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE BOTH)
+
+# Sets a common default architecture for all toolchains. Be sure to update this in GNUToolchain as well!
+if("${SERENITY_ARCH}" STREQUAL "riscv64")
+	add_compile_options(-march=rv64gc)
```
Comment (ADKaster) [inline Toolchain/CMake/ClangToolchain.txt.in:46]: tab char?

**Review by ADKaster — COMMENTED** [review]
some tabs crept in, otherwise this LGTM

**Review by ADKaster — APPROVED** [review]

**Pull request merged**